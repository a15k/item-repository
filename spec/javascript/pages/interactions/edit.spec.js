import Edit from 'pages/interactions/edit';
import Factory from '../../factories';

describe(Edit, () => {
  let props, app;

  beforeEach(() => {
    const collection = Factory.interactionsCollection({ count: 1 });
    app = collection.array[0];

    props = {
      onDone: jest.fn(),
      app,
    };
  });

  it('displays values', () => {
    fetch.mockResponseOnce(JSON.stringify({
      data: {
        name: 'my app', whitelisted_domains: ['foo.com'],
      },
    }));

    const edit = mount(<Edit {...props} />);
    expect(edit).toRender(`input[defaultValue="${props.app.name}"]`);
    edit.find('[name="name"][className="form-control"]').getDOMNode().value = 'my app';
    edit.find('[name="domain"][className="form-control"]').forEach(i => {
      i.getDOMNode().value = 'foo.com';
      i.simulate('blur');
    });
    edit.find('Button[icon="save"]').simulate('click');
    expect(fetch.mock.calls.length).toEqual(1);
    expect(JSON.parse(fetch.mock.calls[0][1].body)).toMatchObject({
      name: 'my app',
      whitelisted_domains: ['foo.com'],
    });
    expect(app.name).toContain('my app');
    edit.unmount();
  });

});
