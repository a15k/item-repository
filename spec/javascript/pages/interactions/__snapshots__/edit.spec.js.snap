// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InteractionEdit matches snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .icon + span {
  margin-left: 0.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 1rem 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 .btn + .btn,
.c2 .btn + .surety-guard,
.c2 .surety-guard + .btn {
  margin-left: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 1rem;
}

<form
  className="c0"
>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="form-control"
      defaultValue="proactive"
      id="name"
      name="name"
      type="text"
    />
  </div>
  <label
    className=""
    htmlFor={undefined}
  >
    <button
      className="c1 btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 icon"
        data-icon="plus"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    Whitelisted Domains
  </label>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
      data-index={0}
    >
      <div
        className="input-group"
      >
        <input
          className="form-control"
          data-index={0}
          defaultValue="jo.info"
          name="domain"
          onBlur={[Function]}
          type="text"
        />
        <div
          className="input-group-append"
        >
          <button
            className="c1 btn btn-secondary"
            data-index={0}
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-trash-alt fa-w-14 icon"
              data-icon="trash-alt"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm416 56v324c0 26.5-21.5 48-48 48H80c-26.5 0-48-21.5-48-48V140c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12zm-272 68c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
    </li>
  </ul>
  <div
    className="form-group"
  >
    <label
      className={undefined}
      htmlFor="copy-on-focus-input1"
    >
      API ID
    </label>
    <input
      className="copy-on-focus form-control"
      id="copy-on-focus-input1"
      onFocus={[Function]}
      readOnly={true}
      value="HLwDJbcYuIk="
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className={undefined}
      htmlFor="copy-on-focus-input2"
    >
      API Token
    </label>
    <input
      className="copy-on-focus form-control"
      id="copy-on-focus-input2"
      onFocus={[Function]}
      readOnly={true}
      value="sti0yiDHyM1S0k3FomGRewSiVbs="
    />
  </div>
  <div
    className="c2"
  >
    <button
      className="c1 btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-left fa-w-10 icon"
        data-icon="chevron-left"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Cancel
      </span>
    </button>
    <button
      className="c1 btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-save fa-w-14 icon"
        data-icon="save"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Save
      </span>
    </button>
  </div>
</form>
`;
